formulaic/transforms/featurehash.py

```diff
@@ -0,0 +1,68 @@
+from hashlib import md5
+from numbers import Number
+from typing import Any, Callable, Dict, Iterable, List, Optional, Union
+
+import numpy as np
+
+from formulaic.materializers.types import FactorValues
+
+from .contrasts import Contrasts, encode_contrasts
+
+
+def md5_to_int(s: str) -> int:
+    return int(md5(s.encode(), usedforsecurity=False).hexdigest(), 16)
+
+
+def hashed(
+    data: Any,
+    levels: int,
+    contrasts: Optional[
+        Union[Contrasts, Dict[str, Iterable[Number]], np.ndarray]
+    ] = None,
+    *,
+    hash_func: Callable[[str], int] = md5_to_int,
+    spans_intercept: bool = False,
+):
+    """
+    Hashes the values of a factor into a fixed number of levels. DUe to the
```

Comment:
nit: Might be worth adding the word "Deterministically"? It is this deterministic nature that allows you to treat this as a regular function rather than a stateful transform.